### PR TITLE
Added Animation component, State component and animation system

### LINKED ISSUE #55: Character animation

### PR DESCRIPTION
**Pull request by eldarht**
#55 Character animation seems to work, but i have not updated movementSystem or bombSystem to change the player state. Was not sure if that would cause conflict.

Should i update the other systems or leave it as it is?

### CODE REVIEW

File: core/src/no/group15/playmagic/utils/assets/AssetEnums.kt [hunk 1]
```diff
@@ -9,6 +9,8 @@ import ktx.freetype.*
 
 enum class GameAssets(override val desc: AssetDescriptor<out Any>) : AssetDesc {
 	BADLOGIC(AssetDescriptor("badlogic.jpg", Texture::class.java)),
+	// Should be a TextureAtlas, but the texture packer did not work for me
```
Comment (DeimosA) [inline core/src/no/group15/playmagic/utils/assets/AssetEnums.kt]: tools extension isn't compatible with lwjgl3 so can't use the online packer

File: core/src/no/group15/playmagic/ecs/Factories.kt [hunk 2]
```diff
@@ -21,6 +21,7 @@ fun engineFactory(injectContext: Context, viewport: Viewport): Engine {
 	engine.addSystem(MovementSystem(1, injectContext, gameMap))
 	engine.addSystem(CollisionSystem(2))
 	engine.addSystem(RenderingSystem(10, viewport, batch))
+	engine.addSystem(AnimationSystem(10))
```
Comment (DeimosA) [inline core/src/no/group15/playmagic/ecs/Factories.kt]: priority should be before rendering I guess, so 9

**Review by DeimosA — CHANGES_REQUESTED** [review]

**Comment by DeimosA** [comment]
upating the other systems should be fine

File: core/src/no/group15/playmagic/ecs/Factories.kt (hunk 2 above)
Comment (eldarht) [reply core/src/no/group15/playmagic/ecs/Factories.kt]: ah, i misinterpreted the description on https://github.com/libgdx/ashley/wiki/How-to-use-Ashley, thought it was for performance and higher number was less important.

**Pull request merged**